### ISSUE TITLE
Floats aren't formatted in query strings

### ISSUE THREAD

**Issue by tomasgrahn** [pymssql/pymssql#326]
Floats aren't formatted in the query string when passing it as a parameter. See the example below:

```python
with _mssql.connect(r"someserver", user="someuser", password="secret!") as conn:
    conn.debug_queries = True
    conn.execute_query("SELECT %f;", 0.0)
```

Result:
```	
#SELECT %f;#

Traceback (most recent call last):
  File "<pyshell#25>", line 3, in <module>
    conn.execute_query("SELECT %f;", 0.0)
  File "_mssql.pyx", line 998, in _mssql.MSSQLConnection.execute_query (_mssql.c:10765)
  File "_mssql.pyx", line 1029, in _mssql.MSSQLConnection.execute_query (_mssql.c:10639)
  File "_mssql.pyx", line 1162, in _mssql.MSSQLConnection.format_and_run_query (_mssql.c:11894)
  File "_mssql.pyx", line 1568, in _mssql.check_cancel_and_raise (_mssql.c:15788)
  File "_mssql.pyx", line 1612, in _mssql.maybe_raise_MSSQLDatabaseException (_mssql.c:16426)
MSSQLDatabaseException: (102, "Incorrect syntax near 'f'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n")
```

Expected result:
```
#SELECT 0.0;#
```

I'm using pymssql 2.1.1 installed using pip on Windows, with the FreeTDS that came bundled with it.

**Comment by rsyring** [comment]
Only %s is fully supported.  That is a design decision, pymssql substitution code will take care of formatting the value into the SQL string based on the type of the value.  

%d has partial support, see #291 for further discussion on that.

**Comment by tomasgrahn** [comment]
From the docs about the execute-method:

> Performs the operation against the database, possibly replacing parameter placeholders with provided values. This should be preferred method of creating SQL commands, instead of concatenating strings manually, what makes a potential of SQL Injection attacks. **This method accepts the same formatting as Python’s builtin string interpolation operator.**

If this is not true then the docs should be changed. Perhaps a future version of pymssql should use a parameter placeholder like SQLite (a question mark regardless of the type of data) to avoid confusion?

**Comment by rsyring** [comment]
Thanks for pointing that out in the docs.  I'll reopen the issue.  

Yes, in hindsight using a question mark would have been preferred.

**Comment by ramiro** [comment]
Extra info: this issue is about _mssql, I don't know if the interpolation
rules/choices described in the DB-API apply here.

I also had to go to the docs to confirm what contain they about this topic
and was surprised to see they announce full support of Python-line
formatting.

On Sat, May 16, 2015 at 11:27 AM, Randy Syring <notifications@github.com>
wrote:

> Reopened #326 <https://github.com/pymssql/pymssql/issues/326>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pymssql/pymssql/issues/326#event-306482993>.
>



-- 
Ramiro Morales
@ramiromorales

**Comment by ramiro** [comment]
Fixed via 7ac0232.
